### PR TITLE
feat(uma2): handle access policies for the UMA Token Granter

### PR DESCRIPTION
**Pull request by tcompiegne**
closes gravitee-io/issues#3861

### CODE REVIEW

File: docs/uma2-api-descriptor.yml [hunk 1]
```diff
@@ -148,6 +148,117 @@ paths:
...
+          content:
+            "application/json":
+              schema:
+                $ref: "#/components/schemas/AccessPolicy"
+        401:
+          $ref: "#/components/responses/Unauthorized"
+        403:
+          $ref: "#/components/responses/Forbidden"
+  "/uma/protection/resource_set/{resource_id}/policies/{policy_id}":
+    put:
```
Comment (lusoalex) [inline docs/uma2-api-descriptor.yml:204]: You missed the GET /uma/protection/resource_set/{resource_id}/policies/{policy_id}

File: docs/uma2-api-descriptor.yml [hunk 2]
```diff
@@ -199,6 +310,15 @@ components:
         application/json:
           schema:
             $ref: "#/components/schemas/Resource_body"
+    # Request for resource access policies
+    AccessPolicy:
+      description: |
+        If the request is successful, the access policy is thereby registered and the authorization server will respond with an HTTP 201 status
+        and include a Location header of the asscociated access policy.
```
[the remark this answers is not in the document]
Comment (tcompiegne) [reply docs/uma2-api-descriptor.yml]: FP

File: gravitee-am-plugins-handlers/gravitee-am-plugins-handlers-policy/src/main/java/io/gravitee/am/plugins/policy/core/impl/PolicyPluginManagerImpl.java [hunk 3]
```diff
@@ -83,7 +83,7 @@ public String getSchema(String policyId) throws IOException {
 
     @Override
     public Policy create(String type, String configuration) {
-        logger.debug("Looking for an reporter provider for [{}]", type);
+        logger.debug("Looking for an policy for [{}]", type);
```
[the remark this answers is not in the document]
Comment (tcompiegne) [reply gravitee-am-plugins-handlers/gravitee-am-plugins-handlers-policy/src/main/java/io/gravitee/am/plugins/policy/core/impl/PolicyPluginManagerImpl.java]: FP

File: gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oauth2/service/granter/uma/UmaTokenGranterTest.java [hunk 4]
```diff
@@ -266,6 +280,7 @@ public void grant_rptNotBelongToSameUser() {
 
     @Test
     public void grant_user_nominalCase() {
+        when(resourceService.findAccessPoliciesByResources(anyList())).thenReturn(Single.just(Collections.emptyList()));
```
Comment (lusoalex) [inline gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oauth2/service/granter/uma/UmaTokenGranterTest.java:283]: Please remove it from the test method to the setup part

Comment (tcompiegne) [reply gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-oidc/src/test/java/io/gravitee/am/gateway/handler/oauth2/service/granter/uma/UmaTokenGranterTest.java]: FP

**Pull request merged**